src/android/CameraPreviewFragment.java

```diff
@@ -1,109 +1,208 @@
-    public void onResume() {
-        super.onResume();
-        camera.open();
+    public void onStart() {
+        super.onStart();
+
+        if (lifecycleRegistry.getCurrentState() == Lifecycle.State.CREATED) {
+            lifecycleRegistry.setCurrentState(Lifecycle.State.STARTED);
+        }
+    }
+
+    public void startCamera() {
+        ListenableFuture<ProcessCameraProvider> cameraProviderFuture = ProcessCameraProvider.getInstance(getActivity());
+        ProcessCameraProvider cameraProvider = null;
+
+        try {
+            cameraProvider = cameraProviderFuture.get();
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            Log.e(TAG, "startCamera: " + e.getMessage());
+        }
+
+        CameraSelector cameraSelector = new CameraSelector.Builder()
+                .requireLensFacing(direction)
+                .build();
+        preview = new Preview.Builder().build();
+        imageCapture = new ImageCapture.Builder().build();
+
+        cameraProvider.unbindAll();
+        camera = cameraProvider.bindToLifecycle(
+                this::getLifecycle,
+                cameraSelector,
+                preview,
+                imageCapture
+        );
+
+        preview.setSurfaceProvider(viewFinder.getSurfaceProvider());
+
+        if (startCameraCallback != null) {
+            startCameraCallback.onCameraStarted();
+        }
+    }
+
+    public void takePicture(boolean useFlash, CameraCallback takePictureCallback) {
+        this.capturePictureCallback = takePictureCallback;
+        camera.getCameraControl().enableTorch(useFlash);
+
+        UUID uuid = UUID.randomUUID();
+
+        File imgFile = new File(
+                getActivity().getBaseContext().getFilesDir(),
+                uuid.toString() + ".jpg"
+        );
+
+        if (imageCapture == null) {
+            imageCapture = new ImageCapture.Builder()
+                    .setCaptureMode(ImageCapture.CAPTURE_MODE_MAXIMIZE_QUALITY)
+                    .setTargetRotation(getActivity().getWindowManager().getDefaultDisplay().getRotation())
+                    .build();
+        }
+
+        ImageCapture.OutputFileOptions outputOptions = new ImageCapture.OutputFileOptions.Builder(imgFile).build();
+        imageCapture.takePicture(
+                outputOptions,
+                ContextCompat.getMainExecutor(getActivity().getApplicationContext()),
+                new ImageCapture.OnImageSavedCallback() {
+                    @Override
+                    public void onImageSaved(@NonNull ImageCapture.OutputFileResults outputFileResults) {
+                        if (camera.getCameraInfo().hasFlashUnit()) {
+                            camera.getCameraControl().enableTorch(false);
+                        }
+
+                        if (imgFile == null) {
+                            capturePictureCallback.onCompleted(new Exception("Unable to save image"), null);
+                        } else {
```

Comment:
Here we could skip the else ? To have a better control-flow ?

```suggestion
                        if (imgFile == null) {
                            capturePictureCallback.onCompleted(new Exception("Unable to save image"), null);
                            return;
                        }
                        // Body of the else
```